src/drivers/github.go

```diff
@@ -24,3 +35,125 @@ func (driver GithubCodeHostingDriver) GetNewPullRequestURL(repository string, br
 func (driver GithubCodeHostingDriver) GetRepositoryURL(repository string) string {
 	return "https://github.com/" + repository
 }
+
+// MergePullRequest merges the pull request through the Github API
+func (driver GithubCodeHostingDriver) MergePullRequest(options MergePullRequestOptions) error {
+	childPullRequestNumbers, err := driver.getPullRequestNumbersAgainst(options.Repository, options.Branch)
+	if err != nil {
+		return err
+	}
+	for _, childPullRequestNumber := range childPullRequestNumbers {
```

Comment:
This method does more than one thing and as a result is too complex. Let's clean it up a bit. 
- extract a helper method called `updatePullRequestsAgainst` that determines the childPullRequestNumbers and then updated these PRs with the parent branch.